gems/core/views.py

```diff
@@ -728,3 +831,38 @@ def get(self, request):
                 "pwd": pwd
             }
         )
+
+
+def get_sms_graph_data():
+    line = list()
+
+    for i in range(6, -1, -1):
+        date = datetime.now() - timedelta(days=i)
+        count = SurveyResult.objects.filter(created_at__day=date.day,
+                                            created_at__month=date.month,
+                                            created_at__year=date.year).count()
+        line.append((-i, count))
+
+    dataset = list()
+    dataset.append(line)
+
+    data = dict()
+    data['heading'] = 'Number of SMSes received past 7 days'
+    data['dataset'] = dataset
+
+    return data
+
+
+def get_graph_data(request):
+
+    if request.method == 'POST':
+        data = json.loads(request.body)
+
+        if 'name' in data:
+
+            if data['name'] == 'SMS_GRAPH':
+                data = get_sms_graph_data()
+
+            return generate_json_response(json.dumps(data))

```

Comment:
I've added this method to accept all the calls for data. Then based on the 'name' it calls the respective method. At this moment we only requesting sms graph data. I guess I can remove the method and then just call the respective method directly by passing the method name as a parameter?
https://github.com/praekelt/gem-survey-tool/blob/develop/gems/core/static/js/graphController.js#L6
That's where I can it and pass the needed data.